sdk_v2/kubeflow/training/types/types.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2024 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from dataclasses import dataclass, field
+from typing import Callable, Dict, List, Optional
+
+from kubeflow.training.constants import constants
+
+
+# Representation for the Training Runtime.
+@dataclass
+class Runtime:
+    name: str
+    phase: str
+    device: str
+    device_count: str
+
+
+# Representation for the TrainJob component.
+@dataclass
+class Component:
+    name: str
+    status: str
+    device: str
+    device_count: str
+    pod_name: str
+
+
+# Representation for the TrainJob.
+# TODO (andreyvelich): Discuss what fields users want to get.
+@dataclass
+class TrainJob:
+    name: str
+    runtime_ref: str
+    creation_timestamp: str
+    components: List[Component]
+    status: Optional[str] = "Unknown"
+
+
+# Configuration for the Lora to configure parameter efficient fine-tuning.
+@dataclass
+class LoraConfig:
+    r: Optional[int] = field(
+        default=None, metadata={"help": "Lora attention dimension"}
+    )
+    lora_alpha: Optional[int] = field(default=None, metadata={"help": "Lora alpha"})
+    lora_dropout: Optional[int] = field(default=None, metadata={"help": "Lora dropout"})
+
+
+@dataclass
+class FineTuningConfig:
+    lora_config: Optional[LoraConfig] = None
+
+
+# Configuration for the Trainer.
+# TODO (andreyvelich): Discuss what values should be on the Trainer.
+@dataclass
+class Trainer:
+    """Trainer configuration.
+    TODO: Add the description
+    """
+
+    func: Optional[Callable] = None
+    func_args: Optional[Dict] = None
+    packages_to_install: Optional[List[str]] = None
+    pip_index_url: str = constants.DEFAULT_PIP_INDEX_URL
+    fine_tuning_config: Optional[FineTuningConfig] = None
+    num_nodes: Optional[int] = None
+    resources_per_node: Optional[dict] = None
+
```

Comment:
@Electronic-Waste How do you view modification of the Trainer APIs ? As you can see, all of these SDK parameters can be transformed to the Trainer CRD definition.

Reply:
@andreyvelich Could `fine_tuning_config` be converted to the Trainer CRD definition? Do you mean that we would pass the fields in `fine_tuning_config` to `Args` or `Env`?

Reply:
Yes, I think so. We want to avoid unnecessary fields in the CRD since we should keep the TrainJob CRD APIs consistent, but the SDK's APIs are easier to change in the future.